internal/auth/authenticators_test.go

```diff
@@ -0,0 +1,143 @@
+	require.NotNil(t, authenticator)
+	require.NoError(t, err)
+
+	req := resty.New().R()
+	err = authenticator.Authenticate(req)
+	require.NoError(t, err)
+	xPubHeadersTestHelper(t, req.Header, key)
+}
+
+func TestXpubAuthenticator_New(t *testing.T) {
+	authenticator, err := auth.NewXpubOnlyAuthenticator(nil)
+	require.Nil(t, authenticator)
+	require.ErrorIs(t, err, auth.ErrBip32ExtendedKey)
+}
+
+func TestAccessKeyAuthenitcator_Authenticate(t *testing.T) {
+	key := privateKeyTestHelper(t)
+	authenticator, err := auth.NewAccessKeyAuthenticator(key)
+	require.NotNil(t, authenticator)
+	require.NoError(t, err)
+
+	req := resty.New().R()
+	err = authenticator.Authenticate(req)
+	require.NoError(t, err)
+	accessKeyHeadersTestHelper(t, req.Header, hex.EncodeToString(key.PubKey().SerializeCompressed()))
+}
+
+func TestAccessKeyAuthenitcator_New(t *testing.T) {
+	authenticator, err := auth.NewAccessKeyAuthenticator(nil)
+	require.Nil(t, authenticator)
+	require.ErrorIs(t, err, auth.ErrEcPrivateKey)
+}
+
+func TestXprivAuthenitcator_Authenticate(t *testing.T) {
+	key := extendedKeyTestHelper(t)
+	authenticator, err := auth.NewXprivAuthenticator(key)
+	require.NotNil(t, authenticator)
+	require.NoError(t, err)
+
+	req := resty.New().R()
+	err = authenticator.Authenticate(req)
+	require.NoError(t, err)
+	xPrivHeadersTestHelper(t, req.Header, key)
+}
+
+func TestXprivAuthenitcator_New(t *testing.T) {
+	authenticator, err := auth.NewXprivAuthenticator(nil)
+	require.Nil(t, authenticator)
+	require.ErrorIs(t, err, auth.ErrBip32ExtendedKey)
+}
+
+func extendedKeyTestHelper(t *testing.T) *bip32.ExtendedKey {
+	t.Helper()
+	key, err := bip32.GenerateHDKey(compat.RecommendedSeedLength)
+	if err != nil {
+		t.Fatalf("test helper - failed to generate HD key from string: %sw", err)
+	}
+	return key
+}
+
+func privateKeyTestHelper(t *testing.T) *ec.PrivateKey {
+	t.Helper()
+	key, err := ec.NewPrivateKey()
+	if err != nil {
+		t.Fatalf("test helper - failed to create private key: %s", err)
+	}
+	return key
+}
+
+func xPubHeadersTestHelper(t *testing.T, h http.Header, key *compat.ExtendedKey) {
+	t.Helper()
+	xPub, err := key.Neuter()
+	if err != nil {
+		t.Fatalf("test helper - failed to get extended public key: %s", err)
+	}
+
+	actualXpub := h["X-Auth-Xpub"]
+	expectedXpub := []string{xPub.String()}
+	require.Equal(t, expectedXpub, actualXpub)
+}
+
+func xPrivHeadersTestHelper(t *testing.T, h http.Header, key *compat.ExtendedKey) {
+	t.Helper()
+	xPub, err := key.Neuter()
+	if err != nil {
+		t.Fatalf("test helper - failed to get extended public key: %s", err)
+	}
+
+	expectedXpub := []string{xPub.String()}
+	expectedHeaders := []string{
+		"X-Auth-Xpub",
+		"X-Auth-Hash",
+		"X-Auth-Nonce",
+		"X-Auth-Time",
+		"X-Auth-Signature",
+	}
+	actualXpub := h["X-Auth-Xpub"]
+	actualHeaders := make([]string, 0, len(expectedHeaders))
+	for k := range h {
+		actualHeaders = append(actualHeaders, k)
+	}
+
+	require.ElementsMatch(t, expectedHeaders, actualHeaders)
```

Comment:
This will check only if headers **contains** expected headers (but could have more), not if it has exactly the expected headers.
```suggestion
	require.Subset(t, actualHeaders, expectedHeaders)
```